contracts/protocol/libs/FeeLib.sol

```diff
@@ -32,15 +32,17 @@ library FeeLib {
 
         // If the token has a custom fee table, find the appropriate percentage
         uint256 priceRangesLength = priceRanges.length;
-        if (priceRangesLength > 0) {
-            for (uint256 i; i < priceRangesLength - 1; ++i) {
-                if (_price <= priceRanges[i]) {
-                    // Return the fee percentage for the matching price range
-                    return feePercentages[i];
+        unchecked {
+            if (priceRangesLength > 0) {
+                for (uint256 i; i < priceRangesLength - 1; ++i) {
+                    if (_price <= priceRanges[i]) {
+                        // Return the fee percentage for the matching price range
+                        return feePercentages[i];
+                    }
                 }
+                // If price exceeds all ranges, use the highest fee percentage
+                return feePercentages[priceRangesLength - 1];
```

Comment:
You can return `feePercentages[i]`, if you declare `uint256 i` before the for loop, saving one subtraction.